Add docs & format Julia files

samuelbelko opened the pull request:
By following the package guide of `Documenter.jl`, I created a folder structure for docs. I can build docs locally by executing make.jl, but don't know how exactly it will interact with github actions defined in `ci.yml`. Also I am not very sure how to make it publish to github pages. As far as I can tell, we need to use [deploydocs function](https://documenter.juliadocs.org/stable/man/hosting/#The-deploydocs-Function) , e.g., like it is used in [POMDPs.jl docs](https://github.com/JuliaPOMDP/POMDPs.jl/blob/master/docs/make.jl) . Maybe someone with more insights could take it from here, or give me hints how to progress further. Thanks!

jbrea commented:
Thanks. You should add a [new workflow file](https://documenter.juliadocs.org/stable/man/hosting/#GitHub-Actions) for deployment (let's try first with `GITHUB_TOKEN`). And yes, you should add the line `deploydocs(repo = "github.com/jbrea/BayesianOptimization.jl.git")` in `make.jl`.

samuelbelko commented:
Thanks! I fixed some docstrings to look better when displayed in docs. I also ran JuliaFormatter on all Julia files & created a config file to set the style to sciml.

I added the line regarding deploydocs in docs/make.jl. But I am still not sure about the workflow file. In `ci.yml` I see a code block `uses: julia-actions/julia-docdeploy` ([here](https://github.com/jbrea/BayesianOptimization.jl/blob/c86125617a29b69b3fc0d2cf6873b0153b1749ae/.github/workflows/ci.yml#L42) ), calling `make.jl`. I think this could do the job but I might not be aware of some details. Assuming that the workflow for building and deploying works properly, the last thing seems to be to setup the source of GitHub pages. ([link to docs](https://documenter.juliadocs.org/stable/man/hosting/#gh-pages-Branch)) 
Please let me know what you think about it & what I can improve. 👍

samuelbelko commented:
Very cool, thanks 👍
